### ISSUE TITLE
An open counter gate has the same move cost as a closed counter gate

### ISSUE THREAD

**Issue by Demivan** [CleverRaven/Cataclysm-DDA#66423]
### Describe the bug

When moving through a counter gate, the character automatically opens it.
This spends extra time, but an opened counter gate has the same move cost as a closed one (200), or a counter without gate. This makes going through a gate slower than going over a regular counter.

### Attach save file

N/A

### Steps to reproduce

1. Spawn a closed counter gate and a regular counter
2. Go through a closed counter gate
3. Character spends 100 moves opening the gate and 161 going through the open gate
4. Go through a regular counter—spend only 161

### Expected behavior

After the character spends time opening a counter gate, going through it should not take the same time as going through a closed one.

### Screenshots

![screenshot_2023-06-24_11-11-47_064698193](https://github.com/CleverRaven/Cataclysm-DDA/assets/2339406/436844a7-9ddc-412e-ab3a-53075339644f)
![screenshot_2023-06-24_11-12-56_430014190](https://github.com/CleverRaven/Cataclysm-DDA/assets/2339406/04367565-e2f3-42d3-90b9-240209e9aec8)

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: 8776427 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm]
]

**Comment by Procyonae** [comment]
/confirm This is very silly, I'll reduce the movecost to 100 but it'll still take about as long to open and move through it as it does to go over the countertop (open close cost is the same hardcoded value for everything as far as I can see)

**Comment by Demivan** [comment]
Yeah, I thought about switching it to have no movement penalty too. But because you  automatically open it, it is still slower.

**Comment by Demivan** [comment]
@Procyonae How about removing automatic opening of counter gates? In addition to making open ones have no movement penalty.

So the character will be going over a counter gate by default. But if you are not in a hurry, you can open the gate and have a tactical tool at your disposal: you can close the gate after yourself, when running from zombies.

**Comment by Procyonae** [comment]
I'll have a look if I can do it with an EOC, there's no other non C++ way to do that though as far as I'm aware

**Comment by Procyonae** [comment]
Eh it's possible but would be jank so I'm going to leave it as is
